lib/molinillo/resolution.rb

```diff
@@ -195,10 +219,8 @@ def unwind_for_conflict
       # @return [Integer] The index to which the resolution should unwind in the
       #   case of conflict.
       def state_index_for_unwind
-        current_requirement = requirement
-        existing_requirement = requirement_for_existing_name(name)
         index = -1
-        [current_requirement, existing_requirement].each do |r|
+        activated.vertex_named(name).requirements.each do |r|
```

Comment:
To be honest, I didn't consider the thinking behind the current "smart" backjumping - this was just a brute force way to fix https://github.com/CocoaPods/Resolver-Integration-Specs/pull/11/ for `ReverseBundlerIndex`. That's pretty niche though, and it was failing before this PR anyway - I'll back this out and we can discuss it in a later PR.